subgraph/README.md

```diff
@@ -36,23 +36,61 @@ Are Unlock's contracts are upgradable, we parse the multiple ABIs that are requi
 yarn prepare:abis
 ```
 
+#### Generating Manifest & Building Subgraph
+
+To build any network. This will generate the network configs, prepare the abis and generate the manifest for the subgraph according to the network configuration.
+
+```sh
+# build & generate
+yarn run build <network-name>
+```
+
 ### Deploying The Subgraph
 
-Deploy the latest subgraph code to the graph node.
+Deploy the latest subgraph code to the hosted-service.
 
 ```sh
 export SUBGRAPH_DEPLOY_KEY=<api-key>
 
-# build
-yarn run build <network-name>
-
 # deploy a single network
 yarn run deploy <network-name>
 
-# deploy all networks
+# deploy all available networks to hosted service
 yarn run deploy-all
 ```
 
+Deploy the latest subgraph code to the studio.
+
+```sh
+export SUBGRAPH_STUDIO_DEPLOY_KEY=<api-key>
+
+# deploy a single network
+yarn run deploy <network-name> --studioName=<Name of studio deployment> --label=<Studio version>
```

Comment:
Unfortunately it would be really messy - I did look into some ways to do it, but it does require updating a lot of variables which could send it out of sync easily. Best way is to use the commit information and some regex.

Reply:
One way to do it is that we add in a seperate field into the networks.subgraph section for `studioVersion`, which we update after every deployment automatically. But this is quite messy. We would need to update versions when we make changes.

Reply:
I would also suggest we add in a section in networks.subgraph with the studio deployment location, and edit the deploy functions to grab that, then it wouldn't require the studioName everytime. I cannot add this in as it is in the types. If you keen on this upgrade, just need `studioEndpoint` in the subgraph section of `NetworkConfig`, I can do the rest.

Reply:
Let me do that!

Reply:
https://github.com/unlock-protocol/unlock/pull/12647

Reply:
Aces! Once its in, I will get the deploy-all working with it. Do you have a list of names for your studio versions? Do you want me to start setting them up and I can transfer after I publish?

Reply:
It's in :)

>  Do you have a list of names for your studio versions? Do you want me to start setting them up and I can transfer after I publish?

Hum, we do have a few but I am a bit lost. Would you be ok for a call next week so we do that together?

Reply:
I was just offering to deploy the ones that haven't been migrated and I could transfer ownership to you after I've published. Let's do the sync and put a bow on this!